go/kbfs/kbfsedits/tlf_history.go

```diff
@@ -386,6 +406,56 @@ func (r *recomputer) processNotification(
 		r.fileEvents[eventFilename] = fileEvent{delete: true}
 	case NotificationDelete:
 		r.fileEvents[eventFilename] = fileEvent{delete: true}
+
+		// We only care about files, so skip dir and sym creates.
+		if notification.FileType != EntryTypeFile {
+			return false
+		}
+
+		// Ignore macOS dotfiles.
+		if ignoreFile(filename) {
+			return false
+		}
+
+		wn, ok := r.byWriter[writer]
+		if !ok {
+			wn = &writerNotifications{writer, nil, nil}
+			r.byWriter[writer] = wn
+		}
+
+		if len(wn.deletes) == maxDeletesPerWriter {
+			// We don't need any more deletes, but we
+			// should continue looking for more edit notifications.
+			return false
+		}
+
+		if hasEvent && event.delete {
+			// It's already been deleted, no need to track it further.
+			return false
+		}
+
+		// If there are no future modifications of this file, then
+		// this delete should be included in the history.
+		for _, files := range r.modifiedFiles {
+			for f := range files {
+				if f == eventFilename {
+					return false
+				}
+			}
+		}
+
+		wn.deletes = append(wn.deletes, notification)
+
+		if len(wn.notifications) == maxEditsPerWriter &&
+			len(wn.deletes) == maxDeletesPerWriter {
+			// We have enough edits and deletes for this user.
+			return true
+		}
+
+		// TODO: limit the number (or time span) of notifications we
```

Comment:
Do we care enough about this to store the most recent n deletes somewhere instead of computing them from the LARGE_NUM most recent operations each time?

How recent is the most recent GC'd revision typically?

Reply:
The computed history is already cached until new data comes in.  We could probably come up with some complicated story to update the deletes separately from the rest of the edit history only when a new delete comes in, but I think this is pretty fast.  Good suggestion, but I probably don't care enough about it to do anything, unless I see some evidence that this is super slow in some cases.